services/slam/fake/slam.go

```diff
@@ -0,0 +1,160 @@
+// Package fake implements a fake base.
+package fake
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"image"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	"go.viam.com/utils"
+	"go.viam.com/utils/artifact"
+
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/rimage"
+	"go.viam.com/rdk/services/slam"
+	"go.viam.com/rdk/spatialmath"
+	rdkutils "go.viam.com/rdk/utils"
+	"go.viam.com/rdk/vision"
+)
+
+var model = resource.NewDefaultModel("fake")
+
+const (
+	internalStateTemplate = "slam/example_cartographer_outputs/internal_state/internal_state_%d.pbstream"
+	maxDataCount          = 16
+	pcdTemplate           = "slam/example_cartographer_outputs/pointcloud/pointcloud_%d.pcd"
+	pngTemplate           = "slam/example_cartographer_outputs/image_map/image_map_%d.png"
+	positionTemplate      = "slam/example_cartographer_outputs/position/position_%d.txt"
+)
+
+func init() {
+	registry.RegisterService(
+		slam.Subtype,
+		model,
+		registry.Service{
+			Constructor: func(
+				ctx context.Context,
+				_ registry.Dependencies,
+				config config.Service,
+				logger golog.Logger,
+			) (interface{}, error) {
+				return &SLAM{Name: config.Name, dataCount: 1}, nil
+			},
+		},
+	)
+}
+
+var _ = slam.Service(&SLAM{})
```

Comment:
I'm surprised this works. Have we not yet added GetPointCloudMap and GetPositionNew to the SLAM service definition?

Reply:
This is fake.SLAM, not slam.SLAM.

Reply:
It is this slam: https://github.com/viamrobotics/rdk/pull/1836/files#r1098790112

Reply:
Oh I see. I think the fake slam needs to fulfill the SLAM service interface and include GetPointCloudMap and GetPositionNew (either in this PR, or in a follow-up).

Reply:
Yep absolutely. I was thinking of adding those endpoints once they are added to `slam/builtin.go` & we have clarity on whether or not the map & internal state endpoints will be streaming. Would that be acceptable?

Reply:
yes, sounds good! do you mind filing a ticket for this?

Reply:
done! https://viam.atlassian.net/browse/RSDK-1865

Reply:
thank you!